host/sgx/create.c

```diff
@@ -874,6 +874,27 @@ oe_result_t oe_sgx_build_enclave(
         &loaded_enclave_pages_size,
         &enclave_size));
 
+    if (context->is_config_id_setting_mandatory && !(props.config.attributes & OE_SGX_FLAGS_KSS))
```

Comment:
If we allow the developer to ignore this case (i.e. if _not_ mandatory _and_ KSS _not_ set) then OE should probably unset `is_config_id_setting_passed` (I would suggest renaming it to `use_config_id`) so downstream functions like `_new_secs()` behave consistently.

Reply:
Some obervations:
- `context->attributes.flags & OE_ENCLAVE_FLAG_SGX_KSS == true` should indicate that both KSS bit is set and the CPU supports the feature after the `if (props.config.attributes & OE_SGX_FLAGS_KSS)` statement.
- We should check only if the `is_config_id_setting_passed` (or `is_config_data_set` as suggested) is set.

Can the two if statements be merged into one as follows (also incorporate @CodeMonkeyLeet'` suggestion)?

```
    if (props.config.attributes & OE_SGX_FLAGS_KSS)
    {
        ...
    }

    if (context->is_config_data_set &&
        !(context->attributes.flags & OE_ENCLAVE_FLAG_SGX_KSS)
    {
        if (!context->config_data.ignore_if_unsupported)
        {
            OE_RAISE_MSG(...);
        }
        else
        {
            context->is_config_data_set = false;
        } 
    }
```